presto-cassandra/src/main/java/com/facebook/presto/cassandra/RandomPartitionerTokenRing.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.cassandra;
+
+import java.math.BigInteger;
+
+import static com.google.common.base.Preconditions.checkArgument;
+
+public final class RandomPartitionerTokenRing
+        implements TokenRing
+{
+    public static final RandomPartitionerTokenRing INSTANCE = new RandomPartitionerTokenRing();
+
+    private static final BigInteger MIN_TOKEN = BigInteger.valueOf(-1);
+    private static final BigInteger MAX_TOKEN = BigInteger.valueOf(2).pow(127);
+    private static final BigInteger TOTAL_TOKEN_COUNT = MAX_TOKEN.subtract(MIN_TOKEN);
+
+    private RandomPartitionerTokenRing() {}
+
+    @Override
+    public double getRingFraction(String start, String end)
+    {
+        return getTokenCountInRange(start, end).doubleValue() / TOTAL_TOKEN_COUNT.doubleValue();
+    }
+
+    @Override
+    public BigInteger getTokenCountInRange(String startToken, String endToken)
+    {
+        BigInteger start = new BigInteger(startToken);
```

Comment:
Though actually, could you use describeOwnership from Partitioner instead of copying this hack from Spark? I'd feel much better about that than having to implement something for every new Partitioner. http://javadox.com/org.apache.cassandra/cassandra-all/2.1.0/org/apache/cassandra/dht/IPartitioner.html#describeOwnership(java.util.List)